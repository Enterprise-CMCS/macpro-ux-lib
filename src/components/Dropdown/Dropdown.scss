@use "../../assets/theme/macbis-colors" as c;
