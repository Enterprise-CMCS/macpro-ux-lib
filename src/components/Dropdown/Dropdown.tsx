import React from "react";

type IntrinsicElements = JSX.IntrinsicElements["select"];

interface DropdownData {
  value: string | number;
  displayString: string;
}

interface Props extends IntrinsicElements {
  data: DropdownData[];
  label: string;
}

/**
 * Dropdown component
 *
 * This component is a wrapper around the HTML `<select>` element and provides the basic functionality of a dropdown along with accessibility and styling.
 * A key enhancement to the standard dropdown is the ability for a user to type to search for an option.
 *
 * @param {DropdownData}  data    Data used to populate the dropdown.
 * @param {string}        id      The id of the dropdown.
 * @param {string}        label   String used to label the dropdown in the UI.
 * @param {string}        name    Name of the dropdown used to identify it in the context of a form.
 */
export const Dropdown: React.FC<Props> = ({
  data,
  id,
  label,
  name,
  ...rest
}) => {
  return (
    <>
      {/*
      Possible TODO:
      These items are outside scope of design, but might be nice to have
      - Error State
      - Default Value
      - Disabled
      - Disable Search
      - Option Group
      - Simple Dropdown - Render without custom styles
      */}

      <label className="usa-label" htmlFor={id}>
        {label}
      </label>
      <div className="usa-combo-box" data-enhanced="true">
        <select
          aria-hidden="true"
          className="usa-select usa-sr-only usa-combo-box__select"
          id=""
          name={name}
          tabIndex={-1}
          {...rest}
        >
          {data.map((option, idx) => (
            <option
              key={`usa-combo-box--option--${id}-${idx}`}
              value={option.value}
            >
              {option.displayString}
            </option>
          ))}
        </select>
        <input
          aria-activedescendant=""
          aria-autocomplete="list"
          aria-controls={`${id}--list`}
          aria-describedby={`${id}--assistiveHint`}
          aria-expanded="false"
          aria-owns={`${id}--list`}
          autoCapitalize="none"
          autoComplete="off"
          className="usa-combo-box__input"
          id={id}
          role="combobox"
          type="text"
        />
        <span className="usa-combo-box__clear-input__wrapper" tabIndex={-1}>
          <button
            aria-label="Clear the select contents"
            className="usa-combo-box__clear-input"
            type="button"
          >
            &nbsp;
          </button>
        </span>
        <span className="usa-combo-box__input-button-separator">&nbsp;</span>
        <span className="usa-combo-box__toggle-list__wrapper" tabIndex={-1}>
          <button
            aria-label="Toggle the dropdown list"
            className="usa-combo-box__toggle-list"
            tabIndex={-1}
            type="button"
          >
            &nbsp;
          </button>
        </span>
        <ul
          aria-labelledby={`${id}-label`}
          className="usa-combo-box__list"
          hidden
          id={`${id}--list`}
          role="listbox"
          tabIndex={-1}
        >
          {data.map((option, idx) => (
            <li
              aria-setsize={data.length}
              aria-posinset={idx + 1}
              aria-selected="false"
              className="usa-combo-box__list-option"
              data-value={option.value}
              id={`${id}--list--option-${idx}`}
              key={`usa-combo-box--item--${id}-${idx}`}
              role="option"
              tabIndex={0}
            >
              {option.displayString}
            </li>
          ))}
        </ul>
        <div className="usa-combo-box__status usa-sr-only" role="status"></div>
        <span className="usa-sr-only" id={`${id}--assistiveHint`}>
          When autocomplete results are available use up and down arrows to
          review and enter to select. Touch device users, explore by touch or
          with swipe gestures.
        </span>
      </div>
    </>
  );
};
