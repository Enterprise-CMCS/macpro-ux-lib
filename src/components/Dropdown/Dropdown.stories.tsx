import React, { useState } from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Dropdown } from "./Dropdown";
import data from "../DropdownInput/data.json";

export default {
  title: "COMPONENTS/Dropdown",
  component: Dropdown,
  args: {
    data,
  },
  argTypes: {
    className: {
      description:
        "A class name that will be applied on the component wrapper `div`.",
      type: { name: "string" },
    },
    data: {
      control: false,
      description:
        "An array of objects used to populate the dropdown. Each object should appear as follows:\n\n `{ value : string | number, displayString: string }`",
    },
    id: {
      description: "The id of the dropdown",
      type: { name: "string", required: true },
    },
    label: {
      description: "String used to label the dropdown in the UI.",
    },
    name: {
      description:
        "Name of the dropdown used to identify it in the context of a form.",
      type: { name: "string", required: true },
    },
    readOnly: {
      description:
        "Sets input field to read-only. Effectively disables type-ahead search.",
    },
    setValue: {
      description:
        "Optionally manage state in this component from a parent by passing state params to `value` and `setValue`",
      control: false,
    },
    value: {
      description:
        "Optionally manage state in this component from a parent by passing state params to `value` and `setValue`",
      control: false,
    },
  },
} as ComponentMeta<typeof Dropdown>;

const Template: ComponentStory<typeof Dropdown> = ({ ...rest }) => {
  const [value, setValue] = useState("");
  return <Dropdown {...rest} value={value} setValue={setValue} />;
};
export const Default = Template.bind({});

Default.args = {
  id: "fruit",
  label: "Select a fruit",
  name: "fruit-dropdown",
};
