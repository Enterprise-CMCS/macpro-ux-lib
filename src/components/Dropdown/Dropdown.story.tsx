import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Dropdown } from "./Dropdown";

export default {
  title: "USWDS/Base/Dropdown",
  component: Dropdown,
  argTypes: {},
  args: {},
} as ComponentMeta<typeof Dropdown>;

const Template: ComponentStory<typeof Dropdown> = ({ ...rest }) => (
  <Dropdown {...rest} />
);

export const PrimaryDropdown = Template.bind({});
PrimaryDropdown.args = {};
