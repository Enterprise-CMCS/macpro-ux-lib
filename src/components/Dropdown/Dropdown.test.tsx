import React from "react";
import { fireEvent as userEvent, render } from "../../test-setup";
import fireEvent from "@testing-library/user-event";

import { Dropdown } from "./Dropdown";
import data from "./data.json";

describe("Test the Dropdown component", () => {
  /*
   * Note:
   *
   * This component uses both the native select elem as well as a custom dropdown.
   * The select element is included for screen readers. This element and a couple
   * others are tagged with a class "usa-sr-only" that render them off to the side
   * of the page. Data should flow to sr-friendly elements correctly and interacting
   * with them should still affect component state.
   */

  let container: HTMLElement;
  let selectElem: Element;
  let inputElem: Element;
  let listElem: Element;

  beforeEach(() => {
    container = render(
      <Dropdown
        data={data}
        id="fruit"
        label="Select a fruit"
        name="fruit-dropdown"
      />
    ).container;
    selectElem = container.getElementsByClassName("usa-combo-box__select")[0];
    inputElem = container.getElementsByClassName("usa-combo-box__input")[0];
    listElem = container.getElementsByClassName("usa-combo-box__list")[0];

    jest
      .spyOn(window, "requestAnimationFrame")
      .mockImplementation((cb: any) => cb());
  });

  it("should render correctly", () => {
    // both dropdowns should exist
    expect(selectElem).toBeInTheDocument();
    expect(inputElem).toBeInTheDocument();
    expect(listElem).toBeInTheDocument();

    // both should contain the same data
    expect(selectElem.children.length).toBe(data.length + 1); // select has a manually added empty option
    expect(listElem.children.length).toBe(data.length);

    // should render collapsed
    expect(listElem).not.toBeVisible();
  });

  it("should expand and collapse", () => {
    // should render collapsed
    expect(listElem).not.toBeVisible();

    // expand on input focus
    userEvent.focus(inputElem);
    expect(listElem).toBeVisible();

    // collapse on toggle click
    const toggleArrow = container.getElementsByClassName(
      "usa-combo-box__toggle-list"
    )[0];
    userEvent.click(toggleArrow);
    expect(listElem).not.toBeVisible();

    // expand on input focus
    userEvent.focus(inputElem);
    expect(listElem).toBeVisible();

    // collapse on input blur
    userEvent.blur(inputElem);
    expect(listElem).not.toBeVisible();
  });

  it("should be able to select elements", () => {
    userEvent.focus(inputElem);

    // select an option
    const apple = container.getElementsByClassName(
      "usa-combo-box__list-option"
    )[0];
    userEvent.click(apple);

    // both should contain the same data
    expect(selectElem).toHaveDisplayValue("Apple");
    expect(inputElem).toHaveDisplayValue("Apple");
  });

  it("should be able to select elements from sr-only select", () => {
    // select an option
    fireEvent.selectOptions(selectElem, "apple");

    // both should contain the same data
    expect(selectElem).toHaveDisplayValue("Apple");
    expect(inputElem).toHaveDisplayValue("Apple");
  });

  it("should be able to clear a selection", () => {
    const clearBtn = container.getElementsByClassName(
      "usa-combo-box__clear-input"
    )[0];
    expect(clearBtn).not.toBeVisible();

    userEvent.focus(inputElem);
    const apple = container.getElementsByClassName(
      "usa-combo-box__list-option"
    )[0];
    userEvent.click(apple);

    // both should contain the same data
    expect(selectElem).toHaveDisplayValue("Apple");
    expect(inputElem).toHaveDisplayValue("Apple");

    expect(clearBtn).toBeVisible();
    userEvent.click(clearBtn);

    // both should be empty
    expect(selectElem).toHaveDisplayValue("");
    expect(inputElem).toHaveDisplayValue("");

    expect(clearBtn).not.toBeVisible();
  });

  it("while input is focused, arrow down should focus on first item", () => {
    userEvent.focus(inputElem);
    userEvent.keyDown(inputElem, { key: "ArrowDown" });

    const apple = container.getElementsByClassName(
      "usa-combo-box__list-option"
    )[0];

    expect(apple).toHaveFocus();
  });

  it("arrows should navigate dropdown list", () => {
    userEvent.focus(inputElem);
    userEvent.keyDown(inputElem, { key: "ArrowDown" });

    const apple = container.getElementsByClassName(
      "usa-combo-box__list-option"
    )[0];
    const apricot = container.getElementsByClassName(
      "usa-combo-box__list-option"
    )[1];

    expect(apple).toHaveFocus();

    // Move down
    userEvent.keyDown(apple, { key: "ArrowDown" });

    expect(apple).not.toHaveFocus();
    expect(apricot).toHaveFocus();

    // Move up
    userEvent.keyDown(apricot, { key: "ArrowUp" });

    expect(apple).toHaveFocus();
    expect(apricot).not.toHaveFocus();

    // Make selection
    userEvent.keyDown(apple, { key: "Enter" });

    expect(inputElem).toHaveDisplayValue("Apple");
    expect(selectElem).toHaveDisplayValue("Apple");
  });

  it("typing in input should filter dropdown list", () => {
    userEvent.focus(inputElem);
    fireEvent.type(inputElem, "apple");

    expect(selectElem.children.length).toBe(65); // select doesn't get filtered
    expect(listElem.children.length).toBe(4);
  });

  it("hover should focus list item", () => {
    userEvent.focus(inputElem);
    const apple = container.getElementsByClassName(
      "usa-combo-box__list-option"
    )[0];
    const apricot = container.getElementsByClassName(
      "usa-combo-box__list-option"
    )[1];

    userEvent.mouseOver(apple);

    expect(apple).toHaveFocus();
    expect(apricot).not.toHaveFocus();

    userEvent.mouseOver(apricot);

    expect(apple).not.toHaveFocus();
    expect(apricot).toHaveFocus();
  });

  it("component snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
