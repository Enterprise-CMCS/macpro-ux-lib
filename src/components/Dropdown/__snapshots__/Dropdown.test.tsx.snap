// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Dropdown component component snapshot 1`] = `
<div>
  <label
    class="usa-label"
  >
    Select a fruit
  </label>
  <div
    class="usa-combo-box"
    data-enhanced="true"
  >
    <select
      aria-hidden="true"
      class="usa-select usa-sr-only usa-combo-box__select"
      id=""
      tabindex="-1"
    >
      <option
        value=""
      >
        Select a fruit
      </option>
      <option
        value="apple"
      >
        Apple
      </option>
      <option
        value="apricot"
      >
        Apricot
      </option>
      <option
        value="avocado"
      >
        Avocado
      </option>
      <option
        value="banana"
      >
        Banana
      </option>
      <option
        value="blackberry"
      >
        Blackberry
      </option>
      <option
        value="blood orange"
      >
        Blood orange
      </option>
      <option
        value="blueberry"
      >
        Blueberry
      </option>
      <option
        value="boysenberry"
      >
        Boysenberry
      </option>
      <option
        value="breadfruit"
      >
        Breadfruit
      </option>
      <option
        value="buddhas hand citron"
      >
        Buddha's hand citron
      </option>
      <option
        value="cantaloupe"
      >
        Cantaloupe
      </option>
      <option
        value="clementine"
      >
        Clementine
      </option>
      <option
        value="crab apple"
      >
        Crab apple
      </option>
      <option
        value="currant"
      >
        Currant
      </option>
      <option
        value="cherry"
      >
        Cherry
      </option>
      <option
        value="custard apple"
      >
        Custard apple
      </option>
      <option
        value="coconut"
      >
        Coconut
      </option>
      <option
        value="cranberry"
      >
        Cranberry
      </option>
      <option
        value="date"
      >
        Date
      </option>
      <option
        value="dragonfruit"
      >
        Dragonfruit
      </option>
      <option
        value="durian"
      >
        Durian
      </option>
      <option
        value="elderberry"
      >
        Elderberry
      </option>
      <option
        value="fig"
      >
        Fig
      </option>
      <option
        value="gooseberry"
      >
        Gooseberry
      </option>
      <option
        value="grape"
      >
        Grape
      </option>
      <option
        value="grapefruit"
      >
        Grapefruit
      </option>
      <option
        value="guava"
      >
        Guava
      </option>
      <option
        value="honeydew melon"
      >
        Honeydew melon
      </option>
      <option
        value="jackfruit"
      >
        Jackfruit
      </option>
      <option
        value="kiwifruit"
      >
        Kiwifruit
      </option>
      <option
        value="kumquat"
      >
        Kumquat
      </option>
      <option
        value="lemon"
      >
        Lemon
      </option>
      <option
        value="lime"
      >
        Lime
      </option>
      <option
        value="lychee"
      >
        Lychee
      </option>
      <option
        value="mandarine"
      >
        Mandarine
      </option>
      <option
        value="mango"
      >
        Mango
      </option>
      <option
        value="mangosteen"
      >
        Mangosteen
      </option>
      <option
        value="marionberry"
      >
        Marionberry
      </option>
      <option
        value="nectarine"
      >
        Nectarine
      </option>
      <option
        value="orange"
      >
        Orange
      </option>
      <option
        value="papaya"
      >
        Papaya
      </option>
      <option
        value="passionfruit"
      >
        Passionfruit
      </option>
      <option
        value="peach"
      >
        Peach
      </option>
      <option
        value="pear"
      >
        Pear
      </option>
      <option
        value="persimmon"
      >
        Persimmon
      </option>
      <option
        value="plantain"
      >
        Plantain
      </option>
      <option
        value="plum"
      >
        Plum
      </option>
      <option
        value="pineapple"
      >
        Pineapple
      </option>
      <option
        value="pluot"
      >
        Pluot
      </option>
      <option
        value="pomegranate"
      >
        Pomegranate
      </option>
      <option
        value="pomelo"
      >
        Pomelo
      </option>
      <option
        value="quince"
      >
        Quince
      </option>
      <option
        value="raspberry"
      >
        Raspberry
      </option>
      <option
        value="rambutan"
      >
        Rambutan
      </option>
      <option
        value="soursop"
      >
        Soursop
      </option>
      <option
        value="starfruit"
      >
        Starfruit
      </option>
      <option
        value="strawberry"
      >
        Strawberry
      </option>
      <option
        value="tamarind"
      >
        Tamarind
      </option>
      <option
        value="tangelo"
      >
        Tangelo
      </option>
      <option
        value="tangerine"
      >
        Tangerine
      </option>
      <option
        value="ugli fruit"
      >
        Ugli fruit
      </option>
      <option
        value="watermelon"
      >
        Watermelon
      </option>
      <option
        value="white currant"
      >
        White currant
      </option>
      <option
        value="yuzu"
      >
        Yuzu
      </option>
    </select>
    <input
      aria-activedescendant=""
      aria-autocomplete="list"
      aria-controls="undefined--list"
      aria-describedby="undefined--assistiveHint"
      aria-expanded="false"
      aria-owns="undefined--list"
      autocapitalize="none"
      autocomplete="off"
      class="usa-combo-box__input"
      role="combobox"
      type="text"
    />
    <span
      class="usa-combo-box__clear-input__wrapper"
      tabindex="-1"
    >
      <button
        aria-label="Clear the select contents"
        class="usa-combo-box__clear-input"
        type="button"
      >
         
      </button>
    </span>
    <span
      class="usa-combo-box__input-button-separator"
    >
       
    </span>
    <span
      class="usa-combo-box__toggle-list__wrapper"
      tabindex="-1"
    >
      <button
        aria-label="Toggle the dropdown list"
        class="usa-combo-box__toggle-list"
        tabindex="-1"
        type="button"
      >
         
      </button>
    </span>
    <ul
      aria-labelledby="undefined-label"
      class="usa-combo-box__list"
      hidden=""
      id="undefined--list"
      role="listbox"
      tabindex="-1"
    >
      <li
        aria-posinset="1"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value=""
        id="undefined--list--option-0"
        role="option"
        tabindex="0"
      >
        Select a fruit
      </li>
      <li
        aria-posinset="2"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="apple"
        id="undefined--list--option-1"
        role="option"
        tabindex="0"
      >
        Apple
      </li>
      <li
        aria-posinset="3"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="apricot"
        id="undefined--list--option-2"
        role="option"
        tabindex="0"
      >
        Apricot
      </li>
      <li
        aria-posinset="4"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="avocado"
        id="undefined--list--option-3"
        role="option"
        tabindex="0"
      >
        Avocado
      </li>
      <li
        aria-posinset="5"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="banana"
        id="undefined--list--option-4"
        role="option"
        tabindex="0"
      >
        Banana
      </li>
      <li
        aria-posinset="6"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="blackberry"
        id="undefined--list--option-5"
        role="option"
        tabindex="0"
      >
        Blackberry
      </li>
      <li
        aria-posinset="7"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="blood orange"
        id="undefined--list--option-6"
        role="option"
        tabindex="0"
      >
        Blood orange
      </li>
      <li
        aria-posinset="8"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="blueberry"
        id="undefined--list--option-7"
        role="option"
        tabindex="0"
      >
        Blueberry
      </li>
      <li
        aria-posinset="9"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="boysenberry"
        id="undefined--list--option-8"
        role="option"
        tabindex="0"
      >
        Boysenberry
      </li>
      <li
        aria-posinset="10"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="breadfruit"
        id="undefined--list--option-9"
        role="option"
        tabindex="0"
      >
        Breadfruit
      </li>
      <li
        aria-posinset="11"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="buddhas hand citron"
        id="undefined--list--option-10"
        role="option"
        tabindex="0"
      >
        Buddha's hand citron
      </li>
      <li
        aria-posinset="12"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="cantaloupe"
        id="undefined--list--option-11"
        role="option"
        tabindex="0"
      >
        Cantaloupe
      </li>
      <li
        aria-posinset="13"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="clementine"
        id="undefined--list--option-12"
        role="option"
        tabindex="0"
      >
        Clementine
      </li>
      <li
        aria-posinset="14"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="crab apple"
        id="undefined--list--option-13"
        role="option"
        tabindex="0"
      >
        Crab apple
      </li>
      <li
        aria-posinset="15"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="currant"
        id="undefined--list--option-14"
        role="option"
        tabindex="0"
      >
        Currant
      </li>
      <li
        aria-posinset="16"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="cherry"
        id="undefined--list--option-15"
        role="option"
        tabindex="0"
      >
        Cherry
      </li>
      <li
        aria-posinset="17"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="custard apple"
        id="undefined--list--option-16"
        role="option"
        tabindex="0"
      >
        Custard apple
      </li>
      <li
        aria-posinset="18"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="coconut"
        id="undefined--list--option-17"
        role="option"
        tabindex="0"
      >
        Coconut
      </li>
      <li
        aria-posinset="19"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="cranberry"
        id="undefined--list--option-18"
        role="option"
        tabindex="0"
      >
        Cranberry
      </li>
      <li
        aria-posinset="20"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="date"
        id="undefined--list--option-19"
        role="option"
        tabindex="0"
      >
        Date
      </li>
      <li
        aria-posinset="21"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="dragonfruit"
        id="undefined--list--option-20"
        role="option"
        tabindex="0"
      >
        Dragonfruit
      </li>
      <li
        aria-posinset="22"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="durian"
        id="undefined--list--option-21"
        role="option"
        tabindex="0"
      >
        Durian
      </li>
      <li
        aria-posinset="23"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="elderberry"
        id="undefined--list--option-22"
        role="option"
        tabindex="0"
      >
        Elderberry
      </li>
      <li
        aria-posinset="24"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="fig"
        id="undefined--list--option-23"
        role="option"
        tabindex="0"
      >
        Fig
      </li>
      <li
        aria-posinset="25"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="gooseberry"
        id="undefined--list--option-24"
        role="option"
        tabindex="0"
      >
        Gooseberry
      </li>
      <li
        aria-posinset="26"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="grape"
        id="undefined--list--option-25"
        role="option"
        tabindex="0"
      >
        Grape
      </li>
      <li
        aria-posinset="27"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="grapefruit"
        id="undefined--list--option-26"
        role="option"
        tabindex="0"
      >
        Grapefruit
      </li>
      <li
        aria-posinset="28"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="guava"
        id="undefined--list--option-27"
        role="option"
        tabindex="0"
      >
        Guava
      </li>
      <li
        aria-posinset="29"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="honeydew melon"
        id="undefined--list--option-28"
        role="option"
        tabindex="0"
      >
        Honeydew melon
      </li>
      <li
        aria-posinset="30"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="jackfruit"
        id="undefined--list--option-29"
        role="option"
        tabindex="0"
      >
        Jackfruit
      </li>
      <li
        aria-posinset="31"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="kiwifruit"
        id="undefined--list--option-30"
        role="option"
        tabindex="0"
      >
        Kiwifruit
      </li>
      <li
        aria-posinset="32"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="kumquat"
        id="undefined--list--option-31"
        role="option"
        tabindex="0"
      >
        Kumquat
      </li>
      <li
        aria-posinset="33"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="lemon"
        id="undefined--list--option-32"
        role="option"
        tabindex="0"
      >
        Lemon
      </li>
      <li
        aria-posinset="34"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="lime"
        id="undefined--list--option-33"
        role="option"
        tabindex="0"
      >
        Lime
      </li>
      <li
        aria-posinset="35"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="lychee"
        id="undefined--list--option-34"
        role="option"
        tabindex="0"
      >
        Lychee
      </li>
      <li
        aria-posinset="36"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="mandarine"
        id="undefined--list--option-35"
        role="option"
        tabindex="0"
      >
        Mandarine
      </li>
      <li
        aria-posinset="37"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="mango"
        id="undefined--list--option-36"
        role="option"
        tabindex="0"
      >
        Mango
      </li>
      <li
        aria-posinset="38"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="mangosteen"
        id="undefined--list--option-37"
        role="option"
        tabindex="0"
      >
        Mangosteen
      </li>
      <li
        aria-posinset="39"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="marionberry"
        id="undefined--list--option-38"
        role="option"
        tabindex="0"
      >
        Marionberry
      </li>
      <li
        aria-posinset="40"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="nectarine"
        id="undefined--list--option-39"
        role="option"
        tabindex="0"
      >
        Nectarine
      </li>
      <li
        aria-posinset="41"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="orange"
        id="undefined--list--option-40"
        role="option"
        tabindex="0"
      >
        Orange
      </li>
      <li
        aria-posinset="42"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="papaya"
        id="undefined--list--option-41"
        role="option"
        tabindex="0"
      >
        Papaya
      </li>
      <li
        aria-posinset="43"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="passionfruit"
        id="undefined--list--option-42"
        role="option"
        tabindex="0"
      >
        Passionfruit
      </li>
      <li
        aria-posinset="44"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="peach"
        id="undefined--list--option-43"
        role="option"
        tabindex="0"
      >
        Peach
      </li>
      <li
        aria-posinset="45"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="pear"
        id="undefined--list--option-44"
        role="option"
        tabindex="0"
      >
        Pear
      </li>
      <li
        aria-posinset="46"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="persimmon"
        id="undefined--list--option-45"
        role="option"
        tabindex="0"
      >
        Persimmon
      </li>
      <li
        aria-posinset="47"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="plantain"
        id="undefined--list--option-46"
        role="option"
        tabindex="0"
      >
        Plantain
      </li>
      <li
        aria-posinset="48"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="plum"
        id="undefined--list--option-47"
        role="option"
        tabindex="0"
      >
        Plum
      </li>
      <li
        aria-posinset="49"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="pineapple"
        id="undefined--list--option-48"
        role="option"
        tabindex="0"
      >
        Pineapple
      </li>
      <li
        aria-posinset="50"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="pluot"
        id="undefined--list--option-49"
        role="option"
        tabindex="0"
      >
        Pluot
      </li>
      <li
        aria-posinset="51"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="pomegranate"
        id="undefined--list--option-50"
        role="option"
        tabindex="0"
      >
        Pomegranate
      </li>
      <li
        aria-posinset="52"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="pomelo"
        id="undefined--list--option-51"
        role="option"
        tabindex="0"
      >
        Pomelo
      </li>
      <li
        aria-posinset="53"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="quince"
        id="undefined--list--option-52"
        role="option"
        tabindex="0"
      >
        Quince
      </li>
      <li
        aria-posinset="54"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="raspberry"
        id="undefined--list--option-53"
        role="option"
        tabindex="0"
      >
        Raspberry
      </li>
      <li
        aria-posinset="55"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="rambutan"
        id="undefined--list--option-54"
        role="option"
        tabindex="0"
      >
        Rambutan
      </li>
      <li
        aria-posinset="56"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="soursop"
        id="undefined--list--option-55"
        role="option"
        tabindex="0"
      >
        Soursop
      </li>
      <li
        aria-posinset="57"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="starfruit"
        id="undefined--list--option-56"
        role="option"
        tabindex="0"
      >
        Starfruit
      </li>
      <li
        aria-posinset="58"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="strawberry"
        id="undefined--list--option-57"
        role="option"
        tabindex="0"
      >
        Strawberry
      </li>
      <li
        aria-posinset="59"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="tamarind"
        id="undefined--list--option-58"
        role="option"
        tabindex="0"
      >
        Tamarind
      </li>
      <li
        aria-posinset="60"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="tangelo"
        id="undefined--list--option-59"
        role="option"
        tabindex="0"
      >
        Tangelo
      </li>
      <li
        aria-posinset="61"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="tangerine"
        id="undefined--list--option-60"
        role="option"
        tabindex="0"
      >
        Tangerine
      </li>
      <li
        aria-posinset="62"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="ugli fruit"
        id="undefined--list--option-61"
        role="option"
        tabindex="0"
      >
        Ugli fruit
      </li>
      <li
        aria-posinset="63"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="watermelon"
        id="undefined--list--option-62"
        role="option"
        tabindex="0"
      >
        Watermelon
      </li>
      <li
        aria-posinset="64"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="white currant"
        id="undefined--list--option-63"
        role="option"
        tabindex="0"
      >
        White currant
      </li>
      <li
        aria-posinset="65"
        aria-selected="false"
        aria-setsize="65"
        class="usa-combo-box__list-option"
        data-value="yuzu"
        id="undefined--list--option-64"
        role="option"
        tabindex="0"
      >
        Yuzu
      </li>
    </ul>
    <div
      class="usa-combo-box__status usa-sr-only"
      role="status"
    />
    <span
      class="usa-sr-only"
      id="undefined--assistiveHint"
    >
      When autocomplete results are available use up and down arrows to review and enter to select. Touch device users, explore by touch or with swipe gestures.
    </span>
  </div>
</div>
`;
